components/script/dom/document.rs

```diff
@@ -1368,6 +1369,13 @@ impl Document {
         loader.load_async(load, listener, self);
     }
 
+    pub fn fetch_async(&self, load: LoadType,
+                       request: RequestInit,
+                       fetch_target: IpcSender<FetchResponseMsg>) {
+        let mut loader = self.loader.borrow_mut();
+        loader.fetch_async(load, request, fetch_target, self);
+    }
```

Comment:
I suspect something's wrong here... there may be a race condition/deadlock between loading scripts and XHR, causing `/XMLHttpRequest/xmlhttprequest-sync-block-scripts.html` to fail.